### ISSUE TITLE
Rule "selector-no-type" false positive with line comment

### ISSUE THREAD

**Issue by celebro** [stylelint/stylelint#930]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where line comments are considered type selectors by `selector-no-type` rule, if they appear before a selector - there's no problem, for instance, if there's a variable definition in between.

> Which rule, if any, is this issue related to?

`selector-no-type`

> What CSS is needed to reproduce this issue?

```css
// Comment
.c {
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "selector-no-type": true,
    }
}
```

> Which version of stylelint are you using?

`5.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `node_modules/.bin/stylelint --config .stylelintrc  example.scss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS line comments.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
example.scss
1:1	Unexpected type selector (selector-no-type)
1:4	Unexpected type selector (selector-no-type)
1:11	Unexpected type selector (selector-no-type)
```

**Comment by davidtheclark** [comment]
@celebro If you can file a PR with at least failing tests that illustrate the need, that would really help speed up this fix.

**Comment by davidtheclark** [comment]
With a31672b799597279dd2d28c75f11a8d3c2c4d973 I created a passing test for this. I believe the problem is that you are not parsing SCSS syntax. Please read http://stylelint.io/user-guide/css-processors/

**Comment by celebro** [comment]
That's true, I apologise. 

Would it make sense to be able to define the syntax from .stylelintrc file, or have some autodetection based on extension? It's cumbersome to configure linting through text editor plugins. And plugin for sublime, for instance, doesn't support setting the syntax at all.
